src/pyhf/pdf.py

```diff
@@ -526,6 +526,12 @@ def _expected_data(self, pars):
         allfac = tensorlib.concatenate(factors + [nom_plus_delta])
 
         newbysample = tensorlib.product(allfac, axis=0)
+        if return_by_sample:
+            batchfirst = tensorlib.einsum('ij...->ji...', newbysample)
+            if self.batch_size is None:
+                return batchfirst[0]
+            return batchfirst
```

Comment:
Codecov is reporting this is uncovered. https://codecov.io/gh/scikit-hep/pyhf/pull/731/diff Can we fix this?

Reply:
should be covered